pmd-java/src/main/java/net/sourceforge/pmd/lang/java/typeresolution/ClassTypeResolver.java

```diff
@@ -97,6 +97,27 @@
     private static final Map<String, Class<?>> PRIMITIVE_TYPES;
     private static final Map<String, String> JAVA_LANG;
 
+    private static ArrayList<Class<?>> primitiveSubTypeOrder = new ArrayList<Class<?>>();
+    private static ArrayList<Class<?>> boxedPrimitivesSubTypeOrder = new ArrayList<>();
```

Comment:
don't init these here. Do it in the static block. Use a local `ArrayList` and then assign these to `Collections.unmodifiableList(tempList)` to make these really immutable